pkg/kube/template.go

```diff
@@ -66,10 +66,10 @@ func (t *Template) Replace() error {
 		replacerFunc = genericReplacement
 	}
 
-	replaceInner(&t.Resource, &t.TemplateData, replacerFunc)
+	replaceInner(&t.Resource, &t.Resource.TemplateData, replacerFunc)
```

Comment:
@jkayani can you take close look here? Not sure how it worked before since `t` doesn't have a `TemplateData` field, its in the Resource struct

Reply:
It's just Go struct embedding right? If struct X has a struct Y, you can access Y's fields via X: https://golang.org/doc/effective_go#embedding. So your change is fine I think but it should also be fine unchanged

Reply:
Oh yeah, that makes sense. I will revert to limit the scope of changes